Run threaded general user objects when not using pthreads

andrsd opened the pull request:
To avoid fails with pthreads.

Refs #11731

<!--
INCLUDE THE FOLLOWING IN THE PR DESCRIPTION
- Explain relevant design information for your change.
- Follow the [Coding Standards](http://mooseframework.org/wiki/CodeStandards/).
- Submit or improve [Test Cases](http://mooseframework.org/wiki/MooseTraining/testing/).
- Reference a specific issue, place "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).
-->

andrsd committed:
Run threaded general user objects only with TBB or OpenMP

To avoid fails with pthreads.

Refs #11731

brianmoose commented on test/tests/userobjects/threaded_general_user_object/tests:
```diff
@@ -6,6 +6,7 @@
     min_threads = 1
     max_threads = 1
     csvdiff = 'th1.csv'
+    tbb = true
```
I don't think we have any civet targets that configure libmesh with `--with-thread-model=tbb`, so these tests will always get skipped. The default thread model is `openmp`. The min clang target uses `pthread`.

andrsd replied:
Yeah. I will need to improve test harness so I can specify the threading model - similarly what we do for compiler, etc.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/218303/) on 5a336fb wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/11961/site)

This comment will be updated on new commits.

andrsd committed:
Fixing fluid_properties module tests

andrsd committed:
Fixing porous_flow tests

So they run only with TBB and openmp

andrsd committed:
Adding threading model specification into test harness

Now, it is possible to restrict test to a threading model.  The old
`tbb` parameter was removed and replaced with `threading` one.  Possible
values are ALL, NONE, TBB, OPENMP, PTHREADS.

andrsd committed:
Fixing navier_stokes module

So it does not run with pthreads

permcody reviewed (approved):
Oh I see. This is pretty important and holding up the current merges to master. This looks good and correct. Thanks!
